cms/djangoapps/api/docs/decisions/001-course-apps.rst

```diff
@@ -0,0 +1,222 @@
+Course Apps API
+_______________
+
+Status
+======
+Proposal
+
+Context
+=======
+
+The new `Course Authoring MFE`_ includes a new UX called "Pages and Resources"
+for configuring different aspects of the course experience such as progress,
+wiki, teams, discussions, etc. all from one place.
+
+Currently, the code for these aspects of the course experience is intermingled
+with the core platform, however, it would be useful to have a way to
+automatically discover the data for driving this page instead of hardcoding it
+such that in the future existing functionality can be split out into an external
+plugin and so that future third-party plugins can also show up here.
+
+This ADR describes how we can use the existing plugin architecture of Open edX
+to automatically discover such apps and expose them via an API. This API will
+be used to list the installed apps that are available for a course and to
+enable/disable these apps using the API.
+
+.. _Course Authoring MFE: https://github.com/edx/frontend-app-course-authoring/
+
+
+Decision
+========
+
+We propose to call such individual course features "Course Apps". They can be
+introduced as a new type of Open edX plugin. Any functionality that can be
+enabled or disabled at a course-level and can be configured by instructors is
+a candidate for being bundled as a Course App.
+
+To be able to show these Course Apps to course admins, they will need to provide
+some bits of metadata, such as a name, a description etc. Additionally we will
+need a common interface for such assps so they can be enabled/disabled
+uniformly.
+
+To do this we can follow the example of existing plugins, [such as Course
+Tabs](https://github.com/edx/edx-platform/blob/636b2ca4c5add531cfce755fdb8965599acd79e0/common/lib/xmodule/xmodule/tabs.py#L24-L243),
+which provide a specific Python class that the plugin can inherit from, or
+implement. The required metadata and features, can be implemented as class
+attributes, and methods on this class.
+
+We can then discover the installed apps using the existing tooling for plugins
+using a subclas of PluginManager designed for this purpose. Here is an example
+for [Course
+Tabs](https://github.com/edx/edx-platform/blob/636b2ca4c5add531cfce755fdb8965599acd79e0/openedx/core/lib/course_tabs.py#L13-L47)
+
+It might not always make sense for an app installed in this way to be
```

Comment:
@xitij2000 by default all apps will be enabled for every course and if the course would not like to use `calculator` there would be a setting in `CourseWaffleFlag` for that course to disable it? and the app will be unavailable for that course?

Reply:
@awaisdar001 The default value for each app is now up to the app itself. Some may default to enabled, others disabled. 

The calculator app cannot be made unavailable, i.e. it's available for all courses. However, it can be disabled for a course. How each app is disabled is also up to the app. For calculator, it's a value stored in the modulestore for the coruse. 
`course.show_calculator`